### PR TITLE
Add Signing To Commit Function

### LINKED ISSUE #274: Allow to automatically sign commits
**Is your feature request related to a problem? Please describe.**
It's not possible to use updatecli in environment that enforce signed commits

**Describe the solution you'd like**
I would to be able to provide a gpg key to signed commits created by updatecli such as

```
    scm:
      github:
        user: updatecli
        email: info@example.com
        owner: updatecli
        repository: updatecli
        token: ...
        username: ...
        branch: main
        gpg: 
          sign: true
          signingKey: xxxx
```

**Describe alternatives you've considered**

**Additional context**
Example of fail commit https://github.com/updatecli/updatecli/pull/272

### PR DESCRIPTION
**Pull request by hi-artem**
# Add gpg signing

Fix #274

Add ability to specify gpg key and passphrase to sign commits. No passphrase will be treated as empty passphrase.

## Test

Tested manually with this config for scm:

```yaml
scms:
  default:
    kind: github
    spec:
      user: "{{ .github.user }}"
      email: "{{ .github.email }}"
      owner: "{{ .github.owner }}"
      repository: "{{ .github.repository }}"
      token: "{{ requiredEnv .github.token }}"
      username: "{{ .github.username }}"
      branch: "{{ .github.branch }}"
      gpg:
        signingKey: |
          -----BEGIN PGP PRIVATE KEY BLOCK-----

          Blablabla... this is secret
          -----END PGP PRIVATE KEY BLOCK-----
        enabled: true
```

Also, added a new test for sign package:

```shell
cd updatecli/pkg/plugins/scms/git/sign
go test
```

### Potential improvement

Any feedback is welcome. I am not familiar with the tool project structure, so if there is a need to change anything in the PR it is cool with me. I am motivated to get it merged.

### CODE REVIEW

**Comment by olblak** [comment]
I wanted to work on this for a quite long time, thanks very much.

File: pkg/plugins/scms/github/scm.go [hunk 1]
```diff
@@ -65,7 +65,13 @@ func (g *Github) Commit(message string) error {
 		return err
 	}
 
-	err = git.Commit(g.Spec.User, g.Spec.Email, commitMessage, g.GetDirectory())
+	gpgSigningKey := g.Spec.GPG.SigningKey
+	gpgPassphrase := g.Spec.GPG.Passphrase
+	if !g.Spec.GPG.Enabled {
```
Comment (olblak) [inline pkg/plugins/scms/github/scm.go:70]: I am not sure to understand the added value of the variable `g.Spec.GPG.Enabled`
I am wondering if we could just remove

**Comment by olblak** [comment]
The pullrquest looks very good, I'll need some time tomorrow to do additionnal testing

File: pkg/plugins/scms/github/scm.go (hunk 1 above)
Comment (hi-artem) [reply pkg/plugins/scms/github/scm.go:70]: No objections removing the `g.Spec.GPG.Enabled`. I was trying to implement the idea from #274 with the key being present in config, but not enabled. Now that I think about it more, there is probably no little value in having it explicitly set to `false` while the key is still defined.

Comment (hi-artem) [reply pkg/plugins/scms/github/scm.go:70]: I will work on changes to remove it.

**Comment by hi-artem** [comment]
@olblak Thank you for reviewing my PR! Looking forward to the testing results.

File: pkg/plugins/scms/github/scm.go (hunk 1 above)
Comment (olblak) [reply pkg/plugins/scms/github/scm.go]: > No objections removing the `g.Spec.GPG.Enabled`. I was trying to implement the idea from #274 with the key being present in config, but not enabled. Now that I think about it more, there is probably no little value in having it explicitly set to `false` while the key is still defined.


Thanks for the clarification, that was my understanding as well. As @dduportal point out, the `enabled` wasn't necessarily useful and while reviewing your code, I couldn't find a motivation to keep it around.

**Comment by olblak** [comment]
I confirmed that I successfully manage to sign commits \o/

**Review by olblak — APPROVED** [review]

**Pull request merged**